www/class/centreon-clapi/centreonDependency.class.php

```diff
@@ -796,24 +851,56 @@ protected function addServiceRelations($depId, $objectToInsert, $relType)
+                );
+            }
+            $sql = "INSERT INTO dependency_serviceParent_relation (dependency_dep_id, host_host_id, service_service_id)
+                VALUES (?, ?, ?)";
             $params = array($depId, $idTab[0], $idTab[1]);
         } elseif ($relType == 'child' && strstr($objectToInsert, ',')) { // service child
-            $sql = "INSERT INTO dependency_serviceChild_relation (dependency_dep_id, host_host_id, service_service_id)
-                VALUES (?, ?, ?)";
             list($host, $service) = explode(",", $objectToInsert);
             $idTab = $this->serviceObj->getHostAndServiceId($host, $service);
             if (!count($idTab)) {
-                throw new CentreonClapiException(sprintf('Could not find service %s on host %s', $service, $host));
+                throw new CentreonClapiException(
+                    sprintf('Could not find service %s on host %s', $depId, $service, $host)
+                );
+            }
+            $dataField = [
+                'dependency_dep_id' => $depId,
+                'host_host_id' => $idTab[0],
+                'service_service_id' => $idTab[1]
+            ];
+            if ($this->isExistingDependency('dependency_serviceChild_relation', $dataField)) {
+                $depName = $this->getDependencyName($depId);
+                throw new CentreonClapiException(
+                    sprintf('Dependency between service %s and host %s already exist on %s', $service, $host, $depName)
+                );
             }
+            $sql = "INSERT INTO dependency_serviceChild_relation (dependency_dep_id, host_host_id, service_service_id)
+                VALUES (?, ?, ?)";
             $params = array($depId, $idTab[0], $idTab[1]);
         } elseif ($relType == 'child') { // host child
-            $sql = "INSERT INTO dependency_hostChild_relation (dependency_dep_id, host_host_id)
-                VALUES (?, ?)";
             $hostObj = new \Centreon_Object_Host($this->dependencyInjector);
             $hostIds = $hostObj->getIdByParameter($hostObj->getUniqueLabelField(), array($objectToInsert));
             if (!count($hostIds)) {
                 throw new CentreonClapiException(sprintf('Could not find host %s', $objectToInsert));
             }
+            $dataField = ['dependency_dep_id' => $depId, 'host_host_id' => $hostIds[0]];
+            if ($this->isExistingDependency('dependency_serviceChild_relation', $dataField)) {
+                $depName = $this->getDependencyName($depId);
+                throw new CentreonClapiException(
+                    sprintf('Service %s already link to the dependency %s', $objectToInsert, $depName)
+                );
+            }
+            $sql = "INSERT INTO dependency_hostChild_relation (dependency_dep_id, host_host_id) VALUES (?, ?)";
             $params = array($depId, $hostIds[0]);
         }
         $this->db->query($sql, $params);
```

Comment:
why not using a prepare / bind / execute statement ? (for other queries into the if elseif too)